gobblin-hive-registration/src/main/java/gobblin/hive/metastore/HiveMetaStoreBasedRegister.java

```diff
@@ -127,52 +133,41 @@ private boolean createDbIfNotExists(IMetaStoreClient client, String dbName) thro
       return false;
     } catch (TException e) {
       throw new IOException("Unable to create Hive database " + dbName, e);
-    } finally {
-      lock.unlock();
     }
   }
 
   @Override
   public boolean createTableIfNotExists(HiveTable table) throws IOException {
-    IMetaStoreClient client = borrowClient();
 
-    HiveLock lock = locks.getTableLock(table.getDbName(), table.getTableName());
-    lock.lock();
-    try {
-      if (client.tableExists(table.getDbName(), table.getTableName())) {
+    try(AutoReturnableObject<IMetaStoreClient> client = this.clientPool.getClient();
+        HiveLock lock = locks.getTableLock(table.getDbName(), table.getTableName())) {
+      if (client.get().tableExists(table.getDbName(), table.getTableName())) {
         return false;
       }
-      client.createTable(HiveMetaStoreUtils.getTable(table));
+      client.get().createTable(HiveMetaStoreUtils.getTable(table));
       return true;
     } catch (TException e) {
       throw new IOException(String.format("Error in creating or altering Hive table %s in db %s", table.getTableName(),
           table.getDbName()), e);
-    } finally {
-      lock.unlock();
-      this.clientPool.returnObject(client);
     }
   }
 
   @Override
   public boolean addPartitionIfNotExists(HiveTable table, HivePartition partition) throws IOException {
-    IMetaStoreClient client = borrowClient();
 
-    HiveLock lock = locks.getTableLock(table.getDbName(), table.getTableName());
-    lock.lock();
-    try {
+    try(AutoReturnableObject<IMetaStoreClient> client = this.clientPool.getClient();
+        HiveLock lock = locks.getTableLock(table.getDbName(), table.getTableName())) {
       try {
-        client.getPartition(table.getTableName(), table.getDbName(), partition.getValues());
+        client.get().getPartition(table.getTableName(), table.getDbName(), partition.getValues());
```

Comment:
No, `getPartition(String db_name, String tbl_name, List<String> part_vals)` in `HiveMetaStoreClient` overrides `getPartition(String tblName, String dbName, List<String> partVals)` in `IMetaStoreClient`.

It's ok if you leave it this way since I'll make the change anyway in my pr